// @vkontakte/vkui/dist/cjs/components/ActionSheet/ActionSheetDropdownDesktop.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _jsxRuntime = require("../../lib/jsxRuntime");

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = require("react");

var _getClassName = require("../../helpers/getClassName");

var _classNames = require("../../lib/classNames");

var _withPlatform = require("../../hoc/withPlatform");

var _withAdaptivity = require("../../hoc/withAdaptivity");

var _dom = require("../../lib/dom");

var ActionSheetDropdownDesktop = /*#__PURE__*/function (_Component) {
  (0, _inherits2.default)(ActionSheetDropdownDesktop, _Component);

  var _super = (0, _createSuper2.default)(ActionSheetDropdownDesktop);

  function ActionSheetDropdownDesktop() {
    var _this;

    (0, _classCallCheck2.default)(this, ActionSheetDropdownDesktop);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
      dropdownStyles: {
        left: '0',
        top: '0',
        opacity: '0',
        pointerEvents: 'none'
      }
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "componentDidMount", function () {
      var _this$props = _this.props,
          toggleRef = _this$props.toggleRef,
          elementRef = _this$props.elementRef,
          popupDirection = _this$props.popupDirection;
      var toggleRect = toggleRef.getBoundingClientRect();
      var elementRect = elementRef.current.getBoundingClientRect();
      var left = toggleRect.left + toggleRect.width - elementRect.width + _this.window.pageXOffset;
      var top;

      if (popupDirection === 'top' || typeof popupDirection === 'function' && popupDirection(elementRef) === 'top') {
        top = toggleRect.top - elementRect.height + _this.window.pageYOffset;
      } else {
        top = toggleRect.top + toggleRect.height + _this.window.pageYOffset;
      }

      _this.setState({
        dropdownStyles: {
          left: left,
          top: top,
          opacity: 1,
          pointerEvents: 'auto'
        }
      });

      setTimeout(function () {
        _this.window.addEventListener('click', _this.handleClickOutside);
      });
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "componentWillUnmount", function () {
      _this.window.removeEventListener('click', _this.handleClickOutside);
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "handleClickOutside", function (e) {
      var dropdownElement = _this.props.elementRef.current;

      if (dropdownElement !== e.target && dropdownElement && !dropdownElement.contains(e.target)) {
        _this.onClose();
      }
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "onClose", function () {
      _this.props.onClose();
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "stopPropagation", function (e) {
      return e.stopPropagation();
    });
    return _this;
  }

  (0, _createClass2.default)(ActionSheetDropdownDesktop, [{
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
          children = _this$props2.children,
          platform = _this$props2.platform,
          elementRef = _this$props2.elementRef,
          toggleRef = _this$props2.toggleRef,
          closing = _this$props2.closing,
          sizeY = _this$props2.sizeY,
          window = _this$props2.window,
          document = _this$props2.document,
          popupDirection = _this$props2.popupDirection,
          restProps = (0, _objectWithoutProperties2.default)(_this$props2, ["children", "platform", "elementRef", "toggleRef", "closing", "sizeY", "window", "document", "popupDirection"]);
      var baseClaseName = (0, _getClassName.getClassName)('ActionSheet', platform);
      return (0, _jsxRuntime.createScopedElement)("div", (0, _extends2.default)({}, restProps, {
        ref: elementRef,
        onClick: this.stopPropagation,
        style: this.state.dropdownStyles,
        vkuiClass: (0, _classNames.classNames)(baseClaseName, 'ActionSheet--desktop', {
          'ActionSheet--closing': this.props.closing
        }, "ActionSheet--sizeY-".concat(sizeY))
      }), children);
    }
  }, {
    key: "window",
    get: function get() {
      return this.props.window;
    }
  }]);
  return ActionSheetDropdownDesktop;
}(_react.Component);

var _default = (0, _withAdaptivity.withAdaptivity)((0, _withPlatform.withPlatform)((0, _dom.withDOM)(ActionSheetDropdownDesktop)), {
  sizeY: true
});

exports.default = _default;
