// @vkontakte/vkui/dist/cjs/components/FormItem/FormItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FormItem = void 0;

var _jsxRuntime = require("../../lib/jsxRuntime");

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = require("react");

var _classNames = require("../../lib/classNames");

var _usePlatform = require("../../hooks/usePlatform");

var _getClassName = require("../../helpers/getClassName");

var _utils = require("../../lib/utils");

var _Subhead = _interopRequireDefault(require("../Typography/Subhead/Subhead"));

var _Caption = _interopRequireDefault(require("../Typography/Caption/Caption"));

var _withAdaptivity = require("../../hoc/withAdaptivity");

var _Removable = require("../Removable/Removable");

var FormItem = (0, _withAdaptivity.withAdaptivity)(function (props) {
  var children = props.children,
      top = props.top,
      bottom = props.bottom,
      status = props.status,
      Component = props.Component,
      sizeY = props.sizeY,
      removable = props.removable,
      onRemove = props.onRemove,
      removePlaceholder = props.removePlaceholder,
      restProps = (0, _objectWithoutProperties2.default)(props, ["children", "top", "bottom", "status", "Component", "sizeY", "removable", "onRemove", "removePlaceholder"]);
  var platform = (0, _usePlatform.usePlatform)();
  var wrappedChildren = (0, _jsxRuntime.createScopedElement)(_react.Fragment, null, (0, _utils.hasReactNode)(top) && (0, _jsxRuntime.createScopedElement)(_Subhead.default, {
    weight: "regular",
    vkuiClass: "FormItem__top"
  }, top), children, (0, _utils.hasReactNode)(bottom) && (0, _jsxRuntime.createScopedElement)(_Caption.default, {
    level: "1",
    weight: "regular",
    vkuiClass: "FormItem__bottom"
  }, bottom));
  return (0, _jsxRuntime.createScopedElement)(Component, (0, _extends2.default)({}, restProps, {
    vkuiClass: (0, _classNames.classNames)((0, _getClassName.getClassName)('FormItem', platform), "FormItem--".concat(status), "FormItem--sizeY-".concat(sizeY), {
      'FormItem--withTop': (0, _utils.hasReactNode)(top)
    })
  }), removable ? (0, _jsxRuntime.createScopedElement)(_Removable.Removable, {
    align: "start",
    onRemove: onRemove,
    removePlaceholder: removePlaceholder
  }, (0, _jsxRuntime.createScopedElement)("div", {
    vkuiClass: "FormItem__removable"
  }, wrappedChildren)) : wrappedChildren);
}, {
  sizeY: true
});
exports.FormItem = FormItem;
FormItem.defaultProps = {
  status: 'default',
  Component: 'div',
  removePlaceholder: 'Удалить'
};
