// @vkontakte/vkui/dist/cjs/components/Chip/Chip.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _jsxRuntime = require("../../lib/jsxRuntime");

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = require("react");

var _icons = require("@vkontakte/icons");

var _utils = require("../../lib/utils");

var _Caption = _interopRequireDefault(require("../Typography/Caption/Caption"));

var Chip = function Chip(props) {
  var value = props.value,
      option = props.option,
      onRemove = props.onRemove,
      removable = props.removable,
      before = props.before,
      after = props.after,
      children = props.children,
      restProps = (0, _objectWithoutProperties2.default)(props, ["value", "option", "onRemove", "removable", "before", "after", "children"]);
  var onRemoveWrapper = (0, _react.useCallback)(function (event) {
    onRemove(event, value);
  }, [onRemove, value]);
  var title = (0, _utils.getTitleFromChildren)(children);
  return (0, _jsxRuntime.createScopedElement)("div", (0, _extends2.default)({
    vkuiClass: "Chip"
  }, restProps), (0, _jsxRuntime.createScopedElement)("div", {
    vkuiClass: "Chip__in"
  }, (0, _utils.hasReactNode)(before) && (0, _jsxRuntime.createScopedElement)("div", {
    vkuiClass: "Chip__before"
  }, before), (0, _jsxRuntime.createScopedElement)(_Caption.default, {
    level: "1",
    weight: "regular",
    vkuiClass: "Chip__content",
    title: title
  }, children), (0, _utils.hasReactNode)(after) && (0, _jsxRuntime.createScopedElement)("div", {
    vkuiClass: "Chip__after"
  }, after), removable && (0, _jsxRuntime.createScopedElement)("div", {
    role: "button",
    tabIndex: 0,
    vkuiClass: "Chip__remove",
    onClick: onRemoveWrapper
  }, (0, _jsxRuntime.createScopedElement)(_icons.Icon16Cancel, {
    fill: "var(--icon_secondary)"
  }))));
};

Chip.defaultProps = {
  removable: true,
  before: null,
  value: '',
  onRemove: _utils.noop
};
var _default = Chip;
exports.default = _default;
