// @vkontakte/vkui/dist/components/FormItem/FormItem.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import { createScopedElement } from "../../lib/jsxRuntime";
import { Fragment } from 'react';
import { classNames } from "../../lib/classNames";
import { usePlatform } from "../../hooks/usePlatform";
import { getClassName } from "../../helpers/getClassName";
import { hasReactNode } from "../../lib/utils";
import Subhead from "../Typography/Subhead/Subhead";
import Caption from "../Typography/Caption/Caption";
import { withAdaptivity } from "../../hoc/withAdaptivity";
import { Removable } from "../Removable/Removable";
export var FormItem = withAdaptivity(function (props) {
  var children = props.children,
      top = props.top,
      bottom = props.bottom,
      status = props.status,
      Component = props.Component,
      sizeY = props.sizeY,
      removable = props.removable,
      onRemove = props.onRemove,
      removePlaceholder = props.removePlaceholder,
      restProps = _objectWithoutProperties(props, ["children", "top", "bottom", "status", "Component", "sizeY", "removable", "onRemove", "removePlaceholder"]);

  var platform = usePlatform();
  var wrappedChildren = createScopedElement(Fragment, null, hasReactNode(top) && createScopedElement(Subhead, {
    weight: "regular",
    vkuiClass: "FormItem__top"
  }, top), children, hasReactNode(bottom) && createScopedElement(Caption, {
    level: "1",
    weight: "regular",
    vkuiClass: "FormItem__bottom"
  }, bottom));
  return createScopedElement(Component, _extends({}, restProps, {
    vkuiClass: classNames(getClassName('FormItem', platform), "FormItem--".concat(status), "FormItem--sizeY-".concat(sizeY), {
      'FormItem--withTop': hasReactNode(top)
    })
  }), removable ? createScopedElement(Removable, {
    align: "start",
    onRemove: onRemove,
    removePlaceholder: removePlaceholder
  }, createScopedElement("div", {
    vkuiClass: "FormItem__removable"
  }, wrappedChildren)) : wrappedChildren);
}, {
  sizeY: true
});
FormItem.defaultProps = {
  status: 'default',
  Component: 'div',
  removePlaceholder: 'Удалить'
};
