// @vkontakte/vkui/dist/cjs/components/FormLayoutGroup/FormLayoutGroup.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _jsxRuntime = require("../../lib/jsxRuntime");

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _getClassName = require("../../helpers/getClassName");

var _classNames = require("../../lib/classNames");

var _usePlatform = require("../../hooks/usePlatform");

var _Removable = require("../Removable/Removable");

var _withAdaptivity = require("../../hoc/withAdaptivity");

var FormLayoutGroup = (0, _withAdaptivity.withAdaptivity)(function (_ref) {
  var children = _ref.children,
      mode = _ref.mode,
      removable = _ref.removable,
      removePlaceholder = _ref.removePlaceholder,
      onRemove = _ref.onRemove,
      sizeY = _ref.sizeY,
      restProps = (0, _objectWithoutProperties2.default)(_ref, ["children", "mode", "removable", "removePlaceholder", "onRemove", "sizeY"]);
  var platform = (0, _usePlatform.usePlatform)();
  var isRemovable = removable && mode === 'horizontal';
  return (0, _jsxRuntime.createScopedElement)("div", (0, _extends2.default)({
    vkuiClass: (0, _classNames.classNames)((0, _getClassName.getClassName)('FormLayoutGroup', platform), "FormLayoutGroup--sizeY-".concat(sizeY), "FormLayoutGroup--".concat(mode))
  }, restProps), isRemovable ? (0, _jsxRuntime.createScopedElement)(_Removable.Removable, {
    align: "start",
    removePlaceholder: removePlaceholder,
    onRemove: onRemove
  }, children) : children);
}, {
  sizeY: true
});
FormLayoutGroup.defaultProps = {
  mode: 'vertical',
  removePlaceholder: 'Удалить'
};
var _default = FormLayoutGroup;
exports.default = _default;
