// @vkontakte/vkui/dist/cjs/hooks/useInsets.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useInsets = useInsets;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _react = require("react");

var _vkBridge = _interopRequireDefault(require("@vkontakte/vk-bridge"));

var initialState = {
  bottom: null,
  top: null,
  left: null,
  right: null
};

function resolveInsets(e) {
  var _e$detail = e.detail,
      type = _e$detail.type,
      data = _e$detail.data;

  switch (type) {
    case 'VKWebAppUpdateConfig':
    case 'VKWebAppUpdateInsets':
      // Устаревшее событие vk-bridge
      var insets = data.insets;

      if (insets) {
        return (0, _objectSpread2.default)((0, _objectSpread2.default)({}, insets), {}, {
          bottom: insets.bottom > 150 ? 0 : insets.bottom // если больше 150 – значит открылась клава и она сама работает как инсет, то есть наш нужно занулить

        });
      }

  }

  return null;
}

_vkBridge.default.subscribe(function (e) {
  var insets = resolveInsets(e);

  if (insets) {
    var htmlElement = window.document.documentElement;

    for (var key in insets) {
      if (insets.hasOwnProperty(key) && (insets[key] > 0 || key === 'bottom')) {
        htmlElement.style.setProperty("--safe-area-inset-".concat(key), "".concat(insets[key], "px"));
      }
    }

    initialState = insets;
  }
});

function useInsets() {
  var _useState = (0, _react.useState)(initialState),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      insets = _useState2[0],
      setInsets = _useState2[1];

  (0, _react.useEffect)(function () {
    function connectListener(e) {
      var insets = resolveInsets(e);

      if (insets) {
        setInsets(insets);
      }
    }

    _vkBridge.default.subscribe(connectListener);

    return function () {
      _vkBridge.default.unsubscribe(connectListener);
    };
  }, []);
  return insets;
}
