// @vkontakte/vkui/dist/cjs/components/ContentCard/ContentCard.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _jsxRuntime = require("../../lib/jsxRuntime");

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _Card = _interopRequireDefault(require("../Card/Card"));

var _Caption = _interopRequireDefault(require("../Typography/Caption/Caption"));

var _Title = _interopRequireDefault(require("../Typography/Title/Title"));

var _Text = _interopRequireDefault(require("../Typography/Text/Text"));

var _Tappable = _interopRequireDefault(require("../Tappable/Tappable"));

var _getClassName = require("../../helpers/getClassName");

var _usePlatform = require("../../hooks/usePlatform");

var _utils = require("../../lib/utils");

var ContentCard = function ContentCard(props) {
  var getRef = props.getRef,
      onClick = props.onClick,
      subtitle = props.subtitle,
      header = props.header,
      text = props.text,
      caption = props.caption,
      className = props.className,
      image = props.image,
      maxHeight = props.maxHeight,
      disabled = props.disabled,
      mode = props.mode,
      style = props.style,
      getRootRef = props.getRootRef,
      restProps = (0, _objectWithoutProperties2.default)(props, ["getRef", "onClick", "subtitle", "header", "text", "caption", "className", "image", "maxHeight", "disabled", "mode", "style", "getRootRef"]);
  var platform = (0, _usePlatform.usePlatform)();
  return (0, _jsxRuntime.createScopedElement)(_Card.default, {
    mode: mode,
    getRootRef: getRootRef,
    vkuiClass: (0, _getClassName.getClassName)('ContentCard', platform),
    style: style,
    className: className
  }, (0, _jsxRuntime.createScopedElement)(_Tappable.default, {
    Component: "div",
    disabled: disabled || typeof onClick !== 'function',
    role: "button",
    onClick: onClick,
    vkuiClass: "ContentCard__tappable"
  }, image && (0, _jsxRuntime.createScopedElement)("img", (0, _extends2.default)({}, restProps, {
    ref: getRef,
    src: image,
    vkuiClass: "ContentCard__img",
    style: {
      maxHeight: maxHeight
    },
    width: "100%"
  })), (0, _jsxRuntime.createScopedElement)("div", {
    vkuiClass: "ContentCard__body"
  }, (0, _utils.hasReactNode)(subtitle) && (0, _jsxRuntime.createScopedElement)(_Caption.default, {
    caps: true,
    vkuiClass: "ContentCard__text",
    weight: "semibold",
    level: "3"
  }, subtitle), (0, _utils.hasReactNode)(header) && (0, _jsxRuntime.createScopedElement)(_Title.default, {
    vkuiClass: "ContentCard__text",
    weight: "semibold",
    level: "3"
  }, header), (0, _utils.hasReactNode)(text) && (0, _jsxRuntime.createScopedElement)(_Text.default, {
    vkuiClass: "ContentCard__text",
    weight: "regular"
  }, text), (0, _utils.hasReactNode)(caption) && (0, _jsxRuntime.createScopedElement)(_Caption.default, {
    vkuiClass: "ContentCard__text",
    weight: "regular",
    level: "1"
  }, caption))));
};

ContentCard.defaultProps = {
  mode: 'shadow'
};
var _default = ContentCard;
exports.default = _default;
