// @vkontakte/vkui/dist/cjs/components/NativeSelect/NativeSelect.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _jsxRuntime = require("../../lib/jsxRuntime");

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = require("react");

var _classNames2 = require("../../lib/classNames");

var _icons = require("@vkontakte/icons");

var _FormField = _interopRequireDefault(require("../FormField/FormField"));

var _withAdaptivity = require("../../hoc/withAdaptivity");

var _getClassName = require("../../helpers/getClassName");

var _Headline = _interopRequireDefault(require("../Typography/Headline/Headline"));

var _Text = _interopRequireDefault(require("../Typography/Text/Text"));

var _platform = require("../../lib/platform");

var _useIsomorphicLayoutEffect = require("../../lib/useIsomorphicLayoutEffect");

var _useEnsuredControl3 = require("../../hooks/useEnsuredControl");

var _useExternRef = require("../../hooks/useExternRef");

var _usePlatform = require("../../hooks/usePlatform");

var NativeSelect = function NativeSelect(_ref) {
  var _classNames;

  var style = _ref.style,
      _ref$defaultValue = _ref.defaultValue,
      defaultValue = _ref$defaultValue === void 0 ? '' : _ref$defaultValue,
      align = _ref.align,
      placeholder = _ref.placeholder,
      children = _ref.children,
      className = _ref.className,
      getRef = _ref.getRef,
      getRootRef = _ref.getRootRef,
      disabled = _ref.disabled,
      sizeX = _ref.sizeX,
      sizeY = _ref.sizeY,
      restProps = (0, _objectWithoutProperties2.default)(_ref, ["style", "defaultValue", "align", "placeholder", "children", "className", "getRef", "getRootRef", "disabled", "sizeX", "sizeY"]);
  var platform = (0, _usePlatform.usePlatform)();

  var _useState = (0, _react.useState)(''),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      title = _useState2[0],
      setTitle = _useState2[1];

  var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2.default)(_useState3, 2),
      notSelected = _useState4[0],
      setNotSelected = _useState4[1];

  var _useEnsuredControl = (0, _useEnsuredControl3.useEnsuredControl)(restProps, {
    defaultValue: defaultValue
  }),
      _useEnsuredControl2 = (0, _slicedToArray2.default)(_useEnsuredControl, 2),
      value = _useEnsuredControl2[0],
      onChange = _useEnsuredControl2[1];

  var selectRef = (0, _useExternRef.useExternRef)(getRef);
  (0, _useIsomorphicLayoutEffect.useIsomorphicLayoutEffect)(function () {
    var selectedOption = selectRef.current.options[selectRef.current.selectedIndex];

    if (selectedOption) {
      setTitle(selectedOption.text);
      setNotSelected(selectedOption.value === '' && placeholder != null);
    }
  }, [value, children]);
  var TypographyComponent = platform === _platform.VKCOM || sizeY === _withAdaptivity.SizeType.COMPACT ? _Text.default : _Headline.default;
  return (0, _jsxRuntime.createScopedElement)(_FormField.default, {
    Component: "label",
    vkuiClass: (0, _classNames2.classNames)((0, _getClassName.getClassName)('Select', platform), (_classNames = {}, (0, _defineProperty2.default)(_classNames, 'Select--not-selected', notSelected), (0, _defineProperty2.default)(_classNames, "Select--align-".concat(align), !!align), (0, _defineProperty2.default)(_classNames, "Select--sizeX--".concat(sizeX), !!sizeX), (0, _defineProperty2.default)(_classNames, "Select--sizeY--".concat(sizeY), !!sizeY), _classNames)),
    className: className,
    style: style,
    getRootRef: getRootRef,
    disabled: disabled,
    after: sizeY === _withAdaptivity.SizeType.COMPACT ? (0, _jsxRuntime.createScopedElement)(_icons.Icon20Dropdown, null) : (0, _jsxRuntime.createScopedElement)(_icons.Icon24Dropdown, null)
  }, (0, _jsxRuntime.createScopedElement)("select", (0, _extends2.default)({}, restProps, {
    disabled: disabled,
    vkuiClass: "Select__el",
    onChange: onChange,
    value: value,
    ref: selectRef
  }), placeholder && (0, _jsxRuntime.createScopedElement)("option", {
    value: ""
  }, placeholder), children), (0, _jsxRuntime.createScopedElement)(TypographyComponent, {
    Component: "div",
    weight: "regular",
    vkuiClass: "Select__container"
  }, (0, _jsxRuntime.createScopedElement)("span", {
    vkuiClass: "Select__title"
  }, title)));
};

var _default = (0, _withAdaptivity.withAdaptivity)(NativeSelect, {
  sizeX: true,
  sizeY: true
});

exports.default = _default;
