// @vkontakte/vkui/dist/cjs/components/CustomSelectOption/CustomSelectOption.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _jsxRuntime = require("../../lib/jsxRuntime");

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _icons = require("@vkontakte/icons");

var _classNames2 = require("../../lib/classNames");

var _utils = require("../../lib/utils");

var _Text = _interopRequireDefault(require("../Typography/Text/Text"));

var CustomSelectOption = function CustomSelectOption(_ref) {
  var _classNames;

  var children = _ref.children,
      hovered = _ref.hovered,
      selected = _ref.selected,
      before = _ref.before,
      after = _ref.after,
      option = _ref.option,
      restProps = (0, _objectWithoutProperties2.default)(_ref, ["children", "hovered", "selected", "before", "after", "option"]);
  var title = typeof children === 'string' ? children : null;
  return (0, _jsxRuntime.createScopedElement)(_Text.default, (0, _extends2.default)({}, restProps, {
    weight: "regular",
    role: "option",
    title: title,
    "aria-selected": selected,
    vkuiClass: (0, _classNames2.classNames)('CustomSelectOption', (_classNames = {}, (0, _defineProperty2.default)(_classNames, 'CustomSelectOption--hover', hovered), (0, _defineProperty2.default)(_classNames, 'CustomSelectOption--selected', !!selected), _classNames))
  }), (0, _utils.hasReactNode)(before) && (0, _jsxRuntime.createScopedElement)("div", {
    vkuiClass: "CustomSelectOption__before"
  }, before), (0, _jsxRuntime.createScopedElement)("div", {
    vkuiClass: "CustomSelectOption__label"
  }, children), (0, _utils.hasReactNode)(after) && (0, _jsxRuntime.createScopedElement)("div", {
    vkuiClass: "CustomSelectOption__after"
  }, after), selected && (0, _jsxRuntime.createScopedElement)("div", {
    vkuiClass: "CustomSelectOption__selectedIcon"
  }, (0, _jsxRuntime.createScopedElement)(_icons.Icon16Done, {
    fill: "var(--accent)"
  })));
};

var _default = CustomSelectOption;
exports.default = _default;
