// @vkontakte/vkui/dist/cjs/components/ChipsSelect/useChipsSelect.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useChipsSelect = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _createForOfIteratorHelper2 = _interopRequireDefault(require("@babel/runtime/helpers/createForOfIteratorHelper"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = require("react");

var _useChipsInput2 = require("../ChipsInput/useChipsInput");

var useChipsSelect = function useChipsSelect(props) {
  var options = props.options,
      filterFn = props.filterFn,
      getOptionLabel = props.getOptionLabel,
      getOptionValue = props.getOptionValue,
      showSelected = props.showSelected;

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      opened = _useState2[0],
      setOpened = _useState2[1];

  var _useState3 = (0, _react.useState)(0),
      _useState4 = (0, _slicedToArray2.default)(_useState3, 2),
      focusedOptionIndex = _useState4[0],
      setFocusedOptionIndex = _useState4[1];

  var _useState5 = (0, _react.useState)(null),
      _useState6 = (0, _slicedToArray2.default)(_useState5, 2),
      focusedOption = _useState6[0],
      setFocusedOption = _useState6[1];

  var _useChipsInput = (0, _useChipsInput2.useChipsInput)(props),
      fieldValue = _useChipsInput.fieldValue,
      selectedOptions = _useChipsInput.selectedOptions,
      chipsInputState = (0, _objectWithoutProperties2.default)(_useChipsInput, ["fieldValue", "selectedOptions"]);

  var handleInputChange = function handleInputChange(e) {
    chipsInputState.handleInputChange(e);

    if (!opened) {
      setOpened(true);
      setFocusedOptionIndex(0);
    }
  };

  var filteredOptions = (0, _react.useMemo)(function () {
    return options.filter(function (option) {
      return filterFn(fieldValue, option, getOptionLabel);
    });
  }, [options, filterFn, fieldValue, getOptionLabel]);
  filteredOptions = (0, _react.useMemo)(function () {
    if (!filteredOptions.length) {
      return filteredOptions;
    }

    var filteredSet = new Set(filteredOptions);
    var selected = selectedOptions.map(function (item) {
      return getOptionValue(item);
    });

    var _iterator = (0, _createForOfIteratorHelper2.default)(filteredSet),
        _step;

    try {
      for (_iterator.s(); !(_step = _iterator.n()).done;) {
        var item = _step.value;

        if (selected.includes(getOptionValue(item))) {
          filteredSet.delete(item);
        }
      }
    } catch (err) {
      _iterator.e(err);
    } finally {
      _iterator.f();
    }

    return (0, _toConsumableArray2.default)(filteredSet);
  }, [showSelected, filteredOptions, selectedOptions]);
  return (0, _objectSpread2.default)((0, _objectSpread2.default)({}, chipsInputState), {}, {
    fieldValue: fieldValue,
    handleInputChange: handleInputChange,
    opened: opened,
    setOpened: setOpened,
    filteredOptions: filteredOptions,
    focusedOptionIndex: focusedOptionIndex,
    setFocusedOptionIndex: setFocusedOptionIndex,
    focusedOption: focusedOption,
    setFocusedOption: setFocusedOption,
    selectedOptions: selectedOptions
  });
};

exports.useChipsSelect = useChipsSelect;
