// @vkontakte/vkui/dist/components/ChipsInput/ChipsInput.js
import _extends from "@babel/runtime/helpers/extends";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import { createScopedElement } from "../../lib/jsxRuntime";
import React, { useState } from 'react';
import FormField from "../FormField/FormField";
import { classNames } from "../../lib/classNames";
import Chip from "../Chip/Chip";
import { noop } from "../../lib/utils";
import { useChipsInput } from "./useChipsInput";
import { withAdaptivity } from "../../hoc/withAdaptivity";
import { prefixClass } from "../../lib/prefixClass";

var ChipsInput = function ChipsInput(props) {
  var style = props.style,
      value = props.value,
      onChange = props.onChange,
      onInputChange = props.onInputChange,
      onKeyDown = props.onKeyDown,
      onBlur = props.onBlur,
      onFocus = props.onFocus,
      children = props.children,
      className = props.className,
      inputValue = props.inputValue,
      getRef = props.getRef,
      getRootRef = props.getRootRef,
      disabled = props.disabled,
      placeholder = props.placeholder,
      tabIndex = props.tabIndex,
      getOptionValue = props.getOptionValue,
      getOptionLabel = props.getOptionLabel,
      getNewOptionData = props.getNewOptionData,
      renderChip = props.renderChip,
      sizeY = props.sizeY,
      after = props.after,
      restProps = _objectWithoutProperties(props, ["style", "value", "onChange", "onInputChange", "onKeyDown", "onBlur", "onFocus", "children", "className", "inputValue", "getRef", "getRootRef", "disabled", "placeholder", "tabIndex", "getOptionValue", "getOptionLabel", "getNewOptionData", "renderChip", "sizeY", "after"]);

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      focused = _useState2[0],
      setFocused = _useState2[1];

  var _useChipsInput = useChipsInput(props),
      fieldValue = _useChipsInput.fieldValue,
      addOptionFromInput = _useChipsInput.addOptionFromInput,
      removeOption = _useChipsInput.removeOption,
      selectedOptions = _useChipsInput.selectedOptions,
      handleInputChange = _useChipsInput.handleInputChange;

  var handleKeyDown = function handleKeyDown(e) {
    if (disabled || restProps.readOnly) {
      e.preventDefault();
      return;
    }

    onKeyDown(e);

    if (e.key === 'Backspace' && !e.defaultPrevented && !fieldValue && selectedOptions.length) {
      removeOption(getOptionValue(selectedOptions[selectedOptions.length - 1]));
      e.preventDefault();
    }

    if (e.key === 'Enter' && !e.defaultPrevented && fieldValue) {
      addOptionFromInput();
      e.preventDefault();
    }
  };

  var handleBlur = function handleBlur(e) {
    if (focused) {
      setFocused(false);
    }

    onBlur(e);
  };

  var handleFocus = function handleFocus(e) {
    if (!focused) {
      setFocused(true);
    }

    onFocus(e);
  };

  var handleChipRemove = function handleChipRemove(_, value) {
    removeOption(value);
  };

  return createScopedElement(FormField, {
    Component: "label",
    getRootRef: getRootRef,
    vkuiClass: classNames('ChipsInput', "ChipsInput--sizeY-".concat(sizeY), {
      'ChipsInput--focused': focused,
      'ChipsInput--withChips': !!selectedOptions.length
    }),
    className: className,
    style: style,
    disabled: disabled,
    after: after
  }, createScopedElement("div", {
    vkuiClass: "ChipsInput__container"
  }, selectedOptions.map(function (option) {
    var value = getOptionValue(option);
    var label = getOptionLabel(option);
    return createScopedElement(React.Fragment, {
      key: value
    }, renderChip({
      option: option,
      value: value,
      label: label,
      onRemove: handleChipRemove,
      disabled: disabled,
      className: prefixClass('ChipsInput__chip')
    }));
  }), createScopedElement("div", {
    vkuiClass: "ChipsInput__input-container"
  }, createScopedElement("input", _extends({
    ref: getRef,
    value: fieldValue,
    autoCapitalize: "none",
    autoComplete: "off",
    autoCorrect: "off",
    spellCheck: false,
    "aria-autocomplete": "list",
    tabIndex: disabled ? null : tabIndex,
    vkuiClass: "ChipsInput__el",
    onChange: handleInputChange,
    onKeyDown: handleKeyDown,
    onFocus: handleFocus,
    onBlur: handleBlur,
    disabled: disabled,
    placeholder: selectedOptions.length ? null : placeholder
  }, restProps)))));
};

export var chipsInputDefaultProps = {
  type: 'text',
  onChange: noop,
  onInputChange: noop,
  onKeyDown: noop,
  onBlur: noop,
  onFocus: noop,
  value: [],
  tabIndex: 0,
  inputValue: '',
  getOptionValue: function getOptionValue(option) {
    return option.value;
  },
  getOptionLabel: function getOptionLabel(option) {
    return option.label;
  },
  getNewOptionData: function getNewOptionData(_, label) {
    return {
      value: label,
      label: label
    };
  },
  renderChip: function renderChip(_ref) {
    var disabled = _ref.disabled,
        value = _ref.value,
        label = _ref.label,
        rest = _objectWithoutProperties(_ref, ["disabled", "value", "label"]);

    return createScopedElement(Chip, _extends({
      value: value,
      removable: !disabled
    }, rest), label);
  }
};
ChipsInput.defaultProps = chipsInputDefaultProps;
export default withAdaptivity(ChipsInput, {
  sizeY: true
});
