// @vkontakte/vkui/dist/lib/dom.js
import _extends from "@babel/runtime/helpers/extends";
import { createScopedElement } from "./jsxRuntime";
import { createContext, useContext } from 'react';
import { canUseDOM } from '@vkontakte/vkjs';
export { canUseDOM, canUseEventListeners, onDOMLoaded } from '@vkontakte/vkjs';
export var getDOM = function getDOM() {
  return {
    window: canUseDOM ? window : null,
    document: canUseDOM ? document : null
  };
};
export var DOMContext = /*#__PURE__*/createContext(getDOM());
export var useDOM = function useDOM() {
  var dom = useContext(DOMContext);
  return {
    window: dom.window || (canUseDOM ? window : null),
    document: dom.document || (canUseDOM ? document : null)
  };
};
export function withDOM(Component) {
  var WithDOM = function WithDOM(props) {
    var dom = useDOM();
    return createScopedElement(Component, _extends({}, props, dom));
  };

  return WithDOM;
}
export function blurActiveElement(document) {
  if (document && document.activeElement) {
    document.activeElement.blur();
  }
}
