// @vkontakte/vkui/dist/lib/utils.js
import { Children } from 'react';
// Является ли переданное значение числовым
export function isNumeric(value) {
  return !isNaN(parseFloat(value)) && isFinite(value);
} // Является ли переданное значение функцией

export function isFunction(value) {
  return typeof value === 'function';
}
export function throttle(fn) {
  var threshhold = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 50;
  var scope = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : window;
  var last;
  var deferTimer;
  return function () {
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    // @ts-ignore
    var context = scope || this;
    var now = Date.now();

    if (last && now < last + threshhold) {
      clearTimeout(deferTimer);
      deferTimer = setTimeout(function () {
        last = now;
        fn.apply(context, args);
      }, threshhold);
    } else {
      last = now;
      fn.apply(context, args);
    }
  };
}
export function debounce(fn, delay) {
  var context = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : window;
  var timeout;
  var args = null;

  var later = function later() {
    return fn.apply(context, args);
  };

  return function () {
    for (var _len2 = arguments.length, a = new Array(_len2), _key2 = 0; _key2 < _len2; _key2++) {
      a[_key2] = arguments[_key2];
    }

    args = a;
    clearTimeout(timeout);
    timeout = setTimeout(later, delay);
  };
}
export function leadingZero(val) {
  var strVal = val.toFixed();

  if (strVal.length === 1) {
    return '0' + strVal;
  }

  return strVal;
}
export function hasReactNode(value) {
  return value !== undefined && value !== false && value !== null && value !== '';
}
export function isPrimitiveReactNode(node) {
  return typeof node === 'string' || typeof node === 'number';
}
export function setRef(element, ref) {
  if (ref) {
    if (typeof ref === 'function') {
      ref(element);
    } else {
      ref.current = element;
    }
  }
}
export function multiRef() {
  for (var _len3 = arguments.length, refs = new Array(_len3), _key3 = 0; _key3 < _len3; _key3++) {
    refs[_key3] = arguments[_key3];
  }

  var current = null;
  return {
    get current() {
      return current;
    },

    set current(element) {
      current = element;
      refs.forEach(function (ref) {
        return setRef(element, ref);
      });
    }

  };
} // eslint-disable-next-line

export var noop = function noop() {};
export function createCustomEvent(window, type, eventInitDict) {
  if (typeof window.CustomEvent !== 'function') {
    var options = eventInitDict || {
      bubbles: false,
      cancelable: false,
      detail: null
    };
    var evt = document.createEvent('CustomEvent');
    evt.initCustomEvent(type, options.bubbles, options.cancelable, options.detail);
    return evt;
  }

  return new window.CustomEvent(type, eventInitDict);
}
export function getTitleFromChildren(children) {
  var label = '';
  Children.map(children, function (child) {
    if (typeof child === 'string') {
      label += child;
    }
  });
  return label;
}
